docs/docs/external-plugins.md

```diff
@@ -298,43 +404,33 @@ This section describes the JSON object representing a serialized entry. An entry
 
 * `name` is a string representing the entry's raw name.
 
-* `methods` is an array specifying the entry's implemented methods. Each element in the array can be a string representing the method's name, or a method-tuple of `[<method_name>, <method_result>]` indicating a prefetched result. The result should have the same format as `<method_name>`'s output in its calling convention. Note that prefetching is a useful way to avoid unnecessary plugin script invocations. If `read` is prefetched, then the entry's `size` attribute will be set to the prefetched content size.
-
-   **Note:** `read`'s method-tuple can also be specified as `["read", <block_readable?>]` where `<block_readable?>` is a Boolean value. Entries that implement `read`'s block-readable calling convention _must_ specify `read` as the method-tuple `["read", true]`. Similarly, entries that implement `read`'s default signature _can_ specify `read` as the method-tuple `["read", false]`, but this is not required. Finally a prefetched `read` result, i.e. a method-tuple of `["read", <content>]`, implies that `<block_readable?>` is false.
+* `methods` is an array specifying the entry's implemented methods. Each element in the array can be a string representing the method's name, or a method-tuple of `[<method_name>, <method_result>]` indicating a prefetched result. The result should have the same format as `<method_name>`'s output in its calling convention. Note that prefetching is a useful way to avoid unnecessary plugin script invocations. Note that only some methods support prefetched results, and some have unique values of `<method_result>` that enable additional functionality. See the `prefetch` section of each method for specific details.
```

Comment:
Naming the method-tuple sections as `prefetch` feels weird even if it is the most common tuple value (I can't help but think "why not call it what it is"). I suggest that this be re-worded to something like

> `methods` is an array specifying the entry's implemented methods. Each element in the array can be a string representing the method's name, or a method-tuple of `[<method_name>, <value>]`, where `<value>` is method-specific. See the `Tuple Value` section of each method for specific details.

I think it's OK to repeat prefetch details in the relevant sections. Or maybe we could add note here indicating that some tuple values represent a prefetched result? Not sure. Either way, I don't think `Prefetch` should be the name of the sections describing the method-tuple value -- we should try to keep that generic to account for `exec`.